File: Backend/models/service.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Float,
    Boolean,
    Text,
    DateTime
)
from datetime import datetime
from db import Base
class Service(Base):
    __tablename__ = "services"

    id = Column(Integer, primary_key=True, index=True)
    service_code = Column(String(50), unique=True, nullable=False)

    service_name = Column(String(255), nullable=False)

    description = Column(Text)
    category = Column(String(255), nullable=False)
    duration_minutes = Column(Integer)

    price = Column(Float, nullable=False)

    status = Column(Boolean, default=True)

    created_at = Column(
        DateTime,
        default=datetime.utcnow
    )

File: flutter_application_1/lib/models/service.dart
class Service {
  final String id;
  final String serviceCode;
  final String serviceName;
  final String category;
  final String? description;
  final int? duration_minutes;
  final double price;
  final bool status;
  final DateTime createdAt;

  Service({
    required this.id,
    required this.serviceCode,
    required this.serviceName,
    required this.category,
    this.description,
    this.duration_minutes,
    required this.price,
    required this.status,
    required this.createdAt,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['id']?.toString() ?? '',
      serviceCode: json['service_code'],
      serviceName: json['service_name'],
      category: json['category'],
      description: json['description'],
      duration_minutes: json['duration_minutes'],
      price: (json['price'] as num).toDouble(),
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'service_code': serviceCode,
      'service_name': serviceName,

      'category': category,
      'description': description,
      'duration_minutes': duration_minutes,
      'price': price,
      'status': status,
    };
  }
}
